ci/playbooks/kuttl/deploy-deps.yaml

```diff
@@ -32,18 +32,28 @@
       loop_control:
         label: "{{ item }}"
 
-    - name: Ensure that the isolated net was configured for crc
-      ansible.builtin.assert:
-        that:
-          - crc_ci_bootstrap_networks_out is defined
-          - "'crc' in crc_ci_bootstrap_networks_out"
-          - "'default' in crc_ci_bootstrap_networks_out['crc']"
+    - name: set facts for further usage within the framework
+      vars:
+        _crc_hostname: "{{ cifmw_crc_hostname | default('crc') }}"
+      block:
+        - name: Ensure that the isolated net was configured for crc
+          ansible.builtin.assert:
+            that:
+              - crc_ci_bootstrap_networks_out is defined
+              - crc_ci_bootstrap_networks_out[_crc_hostname] is defined
+              - crc_ci_bootstrap_networks_out[_crc_hostname]['default'] is defined
 
-    - name: Set facts for further usage within the framework
-      ansible.builtin.set_fact:
-        cifmw_edpm_prepare_extra_vars:
-          NNCP_INTERFACE: "{{ crc_ci_bootstrap_networks_out.crc.default.iface }}"
-          NETWORK_MTU: "{{ crc_ci_bootstrap_networks_out.crc.default.mtu }}"
+        - name: Set facts for further usage within the framework
+          ansible.builtin.set_fact:
+            cifmw_edpm_prepare_extra_vars:
+              NNCP_INTERFACE: "{{ crc_ci_bootstrap_networks_out.crc.default.iface }}"
+              NETWORK_MTU: "{{ crc_ci_bootstrap_networks_out.crc.default.mtu }}"
+              NNCP_DNS_SERVER: >-
+                {{
+                  crc_ci_bootstrap_networks_out[_crc_hostname].default.ip4 |
+                  default(crc_ci_bootstrap_networks_out[_crc_hostname].default.ip) |
+                  replace('/24', '')
```

Comment:
the chagne are relitvly backprotable but the only really concern i have is hardcodign a /24

it would be better to split on / if present and take only the first element or something like that since we should not assume its a /24 CIDR suffix.

that could be addressed in a follow up but that seam to be a little fragile.

it might be good to see if there is a built in filter for getting ips or splitting that you can use to make this more robust. that could be a followup pr as well just highlighting this inbuilt assumption

Reply:
Note the same logic is used at other places in ci-framework. https://github.com/search?q=repo%3Aopenstack-k8s-operators%2Fci-framework%20NNCP_DNS_SERVER&type=code

Reply:
I agree with @SeanMooney . That was done wrong way from the beginning. Let's accept it as it is and replace in some following patch.